extend prop-name-casing to ignore specific names (#2679)
vuejs/eslint-plugin-vue: docs/rules/prop-name-casing.md

```json
{
  "vue/prop-name-casing": ["error",
    "camelCase" | "snake_case",
    {
      "ignoreProps": []
    }
  ]
}
```

- `"camelCase"` (default) ... Enforce property names in `props` to camel case.
- `"snake_case"` ... Enforce property names in `props` to snake case.

> - `ignoreProps` (`string[]`) ... An array of prop names (or patterns) that should be exempt from the case rule check. Use this option for prop names that are intentionally non-compliant, have special naming requirements for Vue components, or are defined by Vue libraries. You can define these entries as literal strings or as regular expressions (written as strings, e.g., `"/^name/"`).

Reviewer: One tiny docs improvement, the explanation for the `ignoreProps` could perhaps be more concise. 
Something like: 

```suggestion
- `ignoreProps` (`string[]`) ... An array of prop names (or patterns) that don't need to follow the specified casing.
```

</eslint-code-block>

### `"ignoreProps": ["foo-bar", "/^_[a-z]+/u"]`

<eslint-code-block :rules="{'vue/prop-name-casing': ['error', 'camelCase', {
ignoreProps: ['foo-bar', '/^_[a-z]+/u'] }]}">

```vue
<script>
export default {
  props: {
    /* ✓ GOOD */
    greetingText: String,

>     foo-bar: String,

Reviewer 2: ```suggestion
    'foo-bar': String,
```

Author: 🤦 of course, sry.
